functions/src/index.ts

```diff
@@ -242,13 +328,52 @@ exports.calculateAqi = functions.pubsub
           nowCastPm25: nowcastPm25.reading,
           aqi: aqi,
         };
+        aqiBufferData = {
+          aqi: aqi,
+          timestamp: FieldValue.serverTimestamp(),
+        };
       }
-    }
+      // Send AQI reading to current-readings to be displayed on the map
+      await firestore.collection('current-reading').doc('pm25').set({
+        lastUpdated: FieldValue.serverTimestamp(),
+        data: currentData,
+      });
 
-    await firestore.collection('current-reading').doc('pm25').set({
-      lastUpdated: FieldValue.serverTimestamp(),
-      data: currentData,
-    });
+      /// / -------- Update the AQI Circular Buffer -------- ////
```

Comment:
Do we want to externalize this to the helper function as above? Most of the comments on this are likely to be very similar to the section on the readings buffer.

Reply:
Let's talk about it quickly during the dev session on 3/3